ocaml/runtime/bigarray.c

```diff
@@ -868,6 +868,51 @@ CAMLprim value caml_ba_uint8_set64(value vb, value vind, value newval)
   return Val_unit;
 }
 
+CAMLprim value caml_ba_uint8_get16_indexed_by_int64(value, value);
+CAMLprim value caml_ba_uint8_get16_indexed_by_int32(value, value);
+CAMLprim value caml_ba_uint8_get16_indexed_by_nativeint(value, value);
+CAMLprim value caml_ba_uint8_get32_indexed_by_int64(value, value);
+CAMLprim value caml_ba_uint8_get32_indexed_by_int32(value, value);
+CAMLprim value caml_ba_uint8_get32_indexed_by_nativeint(value, value);
+CAMLprim value caml_ba_uint8_get64_indexed_by_int64(value, value);
+CAMLprim value caml_ba_uint8_get64_indexed_by_int32(value, value);
+CAMLprim value caml_ba_uint8_get64_indexed_by_nativeint(value, value);
+CAMLprim value caml_ba_uint8_set16_indexed_by_int64(value, value, value);
+CAMLprim value caml_ba_uint8_set16_indexed_by_int32(value, value, value);
+CAMLprim value caml_ba_uint8_set16_indexed_by_nativeint(value, value, value);
+CAMLprim value caml_ba_uint8_set32_indexed_by_int64(value, value, value);
+CAMLprim value caml_ba_uint8_set32_indexed_by_int32(value, value, value);
+CAMLprim value caml_ba_uint8_set32_indexed_by_nativeint(value, value, value);
+CAMLprim value caml_ba_uint8_set64_indexed_by_int64(value, value, value);
+CAMLprim value caml_ba_uint8_set64_indexed_by_int32(value, value, value);
+CAMLprim value caml_ba_uint8_set64_indexed_by_nativeint(value, value, value);
+
+#define CAMLprim_indexed_by(width, name, index_type, val_func)                 \
+  CAMLprim value caml_ba_uint8_get##width##_indexed_by_##name(value vb,        \
+      value vind)                                                              \
+  {                                                                            \
+    index_type idx = val_func(vind);                                           \
+    if (idx != Long_val(Val_long(idx))) caml_array_bound_error();              \
+    return caml_ba_uint8_get##width(vb, Val_long(idx));                        \
+  }                                                                            \
+  CAMLprim value caml_ba_uint8_set##width##_indexed_by_##name(value vb,        \
+      value vind, value newval)                                                \
+  {                                                                            \
+    index_type idx = val_func(vind);                                           \
+    if (idx != Long_val(Val_long(idx))) caml_array_bound_error();              \
+    return caml_ba_uint8_set##width(vb, Val_long(idx), newval);                \
+  }
+
+CAMLprim_indexed_by(16, int64, int64_t, Int64_val)
```

Comment:
Just in case there are any errors in the script that greps for `CAMLprim`, I think I'd prefer this macro not to have `CAMLprim` in its name.